index.js

```diff
@@ -78,32 +78,29 @@ function getRoomIDFromSocket(socket_obj){
 }
 
 // validates socket requests
-function validateSocket(socket_obj){
-    let result = null
-    const room_id = getRoomIDFromSocket(socket_obj)
+function validateSocket(socket, next){
+    const room_id = getRoomIDFromSocketCookie(socket)
     const room = getRoomObjFromID(room_id)
     // adds socket to room if client is missing room and if room exists
     // will drop connections on server restart
-    if (room_id !== null && !socket_obj.rooms.has(room_id) && room !== null){
-        socket_obj.join(room_id)
-        room.addPlayer(socket_obj.id)
-        result = room
+    if (room_id !== null && !socket.rooms.has(room_id) && room !== null){
+        socket.join(room_id)
```

Comment:
still a bit weird
if the function is validate socket
why is it ALSO adding players to rooms and setting rooms?

Reply:
I think these operations should be in the same function but i cant think of a better name

Reply:
linkSocketToRoom